pkg/streamingpromql/functions/common.go

```diff
@@ -34,6 +35,6 @@ func FloatTransformationDropHistogramsFunc(transform func(f float64) float64) In
 		// https://prometheus.io/docs/prometheus/latest/querying/functions
 		pool.PutHPointSlice(seriesData.Histograms)
 		seriesData.Histograms = nil
-		return FloatTransformationFunc(transform)(seriesData, pool)
+		return floatTransformationFunc(transform)(seriesData, pool)
```

Comment:
We should call `floatTransformationFunc(transform)` once (in `FloatTransformationDropHistogramsFunc`, not this inner function) - we don't need to recreate the function each time this method is called.